server/automute.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/pkg/errors"
+)
+
+const (
+	NotifyPropAutomuted = "msteams_automute"
+
+	// This preference tracks if we've enabled automuting for the user by muting all of their connected channels.
+	PreferenceNameAutomuteEnabled = "automute_enabled"
+)
+
+// enableAutomute mutes all of the user's linked channels and DMs and sets a preference to track that automute is
+// enabled. It assumes that the caller has checked that the user is both connected and has their primary platform
+// set to MS Teams.
+func (p *Plugin) enableAutomute(userID string) (bool, error) {
+	return p.setAutomuteEnabledForUser(userID, true)
+}
+
+// disableAutomute mutes all of the user's linked channels and DMs and sets a preference to track that automute is
+// disabled. It assumes that the user is either not connected to MS Teams or has their primary platform set to MM.
```

Comment:
The first part of the comment here sounds like it might be inaccurate? I'm guessing it **unmutes**, but does it also separately track that automute was disabled? Not sure why we'd need to remember that?

Reply:
This does two things:
1. It unmutes all the users' connected channels
2. Sets the `automute_enabled` preference to `false`

`automute_enabled` will always equal "user is connected && primary platform is Teams", so it's kind of redundant, but it being set basically gives me confidence that all of the users' channels are correctly automuted and will stay correctly automuted down the line. It simplifies the followup logic by letting us just check `automute_enabled` when, for example, muting a channel that gets linked

Reply:
Sounds good! Re: the first part of the comment, I wonder if this is meant to read:
```suggestion
// disableAutomute unmutes all of the user's linked channels and DMs and sets a preference to track that automute is
// disabled. It assumes that the user is either not connected to MS Teams or has their primary platform set to MM.
```

Reply:
Oh yeah, good catch. I was constantly mixing up words like mute/unmute/automute/un-automute and automute/automuting when writing this 😛